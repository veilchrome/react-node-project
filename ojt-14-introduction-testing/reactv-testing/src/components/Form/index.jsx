import React, { useState } from "react";

const Form = () => {
  const [name, setName] = useState("");
  const [email, setEmail] = useState("");

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  return (
    <div data-testid="form">
      <form onSubmit={handleSubmit}>
        <label>
          Name:
          <input
            type="text"
            value={name}
            onChange={(e) => setName(e.target.value)}
          />
        </label>
        <label>
          Email:
          <input
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
        </label>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};

export default Form;
