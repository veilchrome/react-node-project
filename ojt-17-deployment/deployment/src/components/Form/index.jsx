// Form.jsx
import React, { useState, useEffect } from "react";
import styled from "styled-components";

const StyledForm = styled.form`
  background-color: #f0f0f0;
  padding: 50px;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  max-width: 500px;
  margin: 20px auto;
  align-items: center;
`;

const InputField = styled.input`
  width: 100%;
  padding: 10px;
  margin-bottom: 10px;
  border: 1px solid #ccc;
  border-radius: 3px;
  font-size: 16px;
`;

const StyledBox = styled.div`
  background-color: #f0f0f0;
  padding: 20px;
  margin-top: 20px;
  border-radius: 5px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  max-width: 500px;
  margin: 0 auto;
`;

const StyledH2 = styled.h2`
  margin: 0 auto;
  margin-top: 20px;
  padding: 10px;
  border-radius: 10px;
  background-color: #1a1a1a;
  color: #fff;
  width: 25%;
  align-items: center;
  text-align: center;
`;

const Form = () => {
  const [formData, setFormData] = useState({
    name: "",
    email: "",
    password: "",
  });

  const [boxContent, setBoxContent] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  useEffect(() => {
    setBoxContent(
      `Name: ${formData.name}, Email: ${formData.email}, Password: ${formData.password}`
    );
  }, [formData]);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData({ ...formData, [name]: value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // Email validation: Check if the email ends with @gmail.com
    if (!formData.email.endsWith("@gmail.com")) {
      setErrorMessage("Please enter a Gmail address.");
      return;
    }
    // Password validation: Check if password contains at least one capital letter
    if (!/[A-Z]/.test(formData.password)) {
      setErrorMessage("Password must contain at least one capital letter.");
      return;
    }
    setErrorMessage(""); // Clear error message if validation passes
  };

  return (
    <>
      <div className="Navbar" data-testid="navbar">
        <StyledH2>Contact Form</StyledH2>
        <StyledForm onSubmit={handleSubmit}>
          <InputField
            type="text"
            name="name"
            placeholder="Enter Name"
            value={formData.name}
            onChange={handleChange}
            aria-label="Name:"
          />
          <InputField
            type="email"
            name="email"
            placeholder="Enter Email"
            value={formData.email}
            onChange={handleChange}
            aria-label="Email:"
          />
          <InputField
            type="password"
            name="password"
            placeholder="Enter Password"
            value={formData.password}
            onChange={handleChange}
            aria-label="Password:"
          />
          <p>*Please Input Google mails for the Email</p>
          <p>*Passowrd atleast have 1 Capital Letter</p>
          <p>Test testing111</p>
          {errorMessage && <p style={{ color: "red" }}>{errorMessage}</p>}
          <button type="submit">Submit</button>
        </StyledForm>
        <StyledBox>
          <h2>Form Data:</h2>
          <p>{boxContent}</p>
        </StyledBox>
      </div>
    </>
  );
};

export default Form;
