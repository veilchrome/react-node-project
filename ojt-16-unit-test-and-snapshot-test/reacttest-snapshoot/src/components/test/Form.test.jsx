import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import Form from "../Form";
import "@testing-library/jest-dom";

describe("Form Component", () => {
  test("renders form with name and email inputs", () => {
    const { container } = render(<Form />);
    expect(container).toMatchSnapshot();
  });

  test("submits form with correct values", async () => {
    const { getByLabelText, getByText, container } = render(<Form />);
    const nameInput = getByLabelText("Name:");
    const emailInput = getByLabelText("Email:");
    const passwordInput = getByLabelText("Password:");
    const submitButton = getByText("Submit");

    // Fill out form inputs
    fireEvent.change(nameInput, { target: { value: "Selamet Saputra" } });
    fireEvent.change(emailInput, {
      target: { value: "selametsaputra123@gmail.com" },
    });
    fireEvent.change(passwordInput, { target: { value: "Password12345" } });

    // Submit form
    fireEvent.click(submitButton);

    // Wait for form data to be updated
    await waitFor(() => {
      expect(
        getByText(
          "Name: Selamet Saputra, Email: selametsaputra123@gmail.com, Password: Password12345"
        )
      ).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
