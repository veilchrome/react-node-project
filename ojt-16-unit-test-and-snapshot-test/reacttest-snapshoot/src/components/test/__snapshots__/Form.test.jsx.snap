// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders form with name and email inputs 1`] = `
<div>
  <div
    class="Navbar"
    data-testid="navbar"
  >
    <h2
      class="sc-hLQSwg etDQsz"
    >
      Contact Form
    </h2>
    <form
      class="sc-beySPh foRdIa"
    >
      <input
        aria-label="Name:"
        class="sc-guDLey imaPpo"
        name="name"
        placeholder="Enter Name"
        type="text"
        value=""
      />
      <input
        aria-label="Email:"
        class="sc-guDLey imaPpo"
        name="email"
        placeholder="Enter Email"
        type="email"
        value=""
      />
      <input
        aria-label="Password:"
        class="sc-guDLey imaPpo"
        name="password"
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <p>
        *Please Input Google mails for the Email
      </p>
      <p>
        *Passowrd atleast have 1 Capital Letter
      </p>
      <p>
        Test testing111
      </p>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="sc-dmyCSP kOYcjl"
    >
      <h2>
        Form Data:
      </h2>
      <p>
        Name: , Email: , Password: 
      </p>
    </div>
  </div>
</div>
`;

exports[`Form Component submits form with correct values 1`] = `
<div>
  <div
    class="Navbar"
    data-testid="navbar"
  >
    <h2
      class="sc-hLQSwg etDQsz"
    >
      Contact Form
    </h2>
    <form
      class="sc-beySPh foRdIa"
    >
      <input
        aria-label="Name:"
        class="sc-guDLey imaPpo"
        name="name"
        placeholder="Enter Name"
        type="text"
        value="Selamet Saputra"
      />
      <input
        aria-label="Email:"
        class="sc-guDLey imaPpo"
        name="email"
        placeholder="Enter Email"
        type="email"
        value="selametsaputra123@gmail.com"
      />
      <input
        aria-label="Password:"
        class="sc-guDLey imaPpo"
        name="password"
        placeholder="Enter Password"
        type="password"
        value="Password12345"
      />
      <p>
        *Please Input Google mails for the Email
      </p>
      <p>
        *Passowrd atleast have 1 Capital Letter
      </p>
      <p>
        Test testing111
      </p>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      class="sc-dmyCSP kOYcjl"
    >
      <h2>
        Form Data:
      </h2>
      <p>
        Name: Selamet Saputra, Email: selametsaputra123@gmail.com, Password: Password12345
      </p>
    </div>
  </div>
</div>
`;
